src/snowflake/cli/api/utils/rendering.py

```diff
@@ -88,22 +94,151 @@ def jinja_render_from_file(
         print(rendered_result)
 
 
-class _AttrGetter:
+def _is_key_in_environ(key: str):
+    return key.upper() in os.environ or key in os.environ
+
+
+def _get_value_from_environ(key: str):
+    if key.upper() in os.environ:
+        return os.environ[key.upper()]
+    return os.environ[key]
```

Comment:
I think we should do key first, and if not found, try upper.

Reply:
I'm not sure we should be trying to retrofit case-insensitivity onto a case-sensitive platform.

See https://docs.python.org/3/library/os.html#os.environ -- it already takes care of this on relevant platforms. So we can use `key` as-is.

Reply:
even better.

Reply:
Should we consider using best practices for naming environment variables? For example I'm referring to
https://google.github.io/styleguide/shellguide.html#constants-and-environment-variable-names

Honestly, lower-case env vars looks strange and uncommon. I'm fine with doing a fallback to key but let's not force users to use CAPS names in snowflake.yml only to be able to write `export CAPS` instead of `export caps`. WDYT?

Reply:
I'm sure CLI supports the latter on Windows (if not, I'm not sure how that ended up being the case!).

I guess the point is, why should we be opinionated about env var case sensitivity? It's already "solved" at the OS level. It's worth trying out comparables and seeing how they react to case sensitivity, as that will likely be the users' expectation. If other tooling makes things case insensitive, I'm OK with the approach here, but that isn't the case with what I'm seeing, e.g.

- Github actions uses case sensitive env vars: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Reply:
> I  would guess that CLI supports the latter on Windows (if not, I'm curious how that ended up being the case!).

It's not CLI, it's the platform that supports it due to case-insensitivity. For case-sensitive platforms (which are the only concern) we expect variables to be upper case. 

Ok, let's use the keys. If someone want to follow best practices they still are free to do so using upper-case names. We just have to document this properly.